### PR TITLE
Bug 1834254: fetch eventSource icon based on type and update icon for ping

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-3828

**Analysis / Root cause**: 
- Icon's for eventSources is fetched based on resolved reference for kind i.e with `apiVersion/Group/kind` and with dynamic sources this may change going forward as apiVersion/group is fetched from CRDs for sources.
- default icon is shown for PingSource
 

**Solution Description**: 
- Icons for eventSources are fetched based on kind for sources
- Updates icon for CronJob to Svg and same is used for ping Source

**Screen shots / Gifs for design review**: 
Screenshot 2020-05-11 at 4 35 28 PM

eventSourceIconGif

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

cc: @openshift/team-devconsole-ux @serenamarie125

### CODE REVIEW

File: frontend/packages/knative-plugin/src/utils/get-knative-icon.ts [hunk 1]
```diff
@@ -11,19 +11,21 @@ import {
...
   EventSourceCamelModel,
   EventSourceKafkaModel,
+  EventSourcePingModel,
 } from '../models';
 
 export const getKnativeEventSourceIcon = (kind: string): string => {
-  switch (kind) {
-    case referenceForModel(EventSourceApiServerModel):
+  switch (kindForReference(kind)) {
+    case EventSourceApiServerModel.kind:
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/utils/get-knative-icon.ts:19]: Why did you change it to not use `referenceForModel`? A different version of the same event source can probably have a different icon.

File: frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts [hunk 2]
```diff
@@ -205,7 +205,7 @@ export const useEventSourceList = (namespace: string): NormalizedEventSources =>
         ...(checkAccessVal && {
           [eventSourceModel.kind]: {
             name: eventSourceModel.kind,
-            iconUrl: getKnativeEventSourceIcon(referenceForModel(eventSourceModel)),
+            iconUrl: getKnativeEventSourceIcon(eventSourceModel.kind),
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts:208]: I think referenceForModel was a better approach.

File: frontend/packages/knative-plugin/src/utils/get-knative-icon.ts (hunk 1 above)
Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/utils/get-knative-icon.ts:19]: `referenceForModel` returns `sources.knative.dev~v1alpha1~PingSource` and i don't want check with to associated with version/apiGroup as we currently we are fetching these dynamically so rather should be based on Kind.

File: frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts (hunk 2 above)
Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts:208]: same as above, TBH doesn't matter here as it's handled above
